packages/components/src/card/README.md

```diff
@@ -0,0 +1,97 @@
+# Card
+
+Card provides a flexible and extensible content container.
+
+## Usage
+
+```jsx
+import { Card, CardBody } from '@wordpress/components';
+
+const Example = () => (
+	<Card>
+		<CardBody>...</CardBody>
+	</Card>
+);
+```
+
+## Props
+
+<!-- prettier-ignore -->
+Name | Type | Default | Description
+--- | --- | --- | ---
+`size` | `string` | `medium` | Determines the amount of padding within the card.
+`variant` | `string` | `default` | Determines the style of the card.
```

Comment:
I have some thoughts about `variant`:

Is there benefit to having a named `variant="default"` prop as opposed to omitting it?

Are these three possible values tightly related? Put another way, are we intentionally discouraging usage of `raised` + `borderless`? I can understand that the values `raised`, and `borderless` may be related, but I want to clarify whether that was intentional or incidental?

The name `variant` seems imprecise. For example, couldn't `shady` be a `variant`? Should `variant` become something like `border-style = border | raised | none`? When we have props that are imprecise and don't discreetly work with a feature, it may lead to `variant="borderless-shady-rounded" which isn't good 🙂

Reply:
@sirreal Ah yes! Thanks for bringing this up. I wasn't sure how to name/handle these, since it doesn't look like we use variants in the current component set.

In the case of `<Button />`, a lot of the props are `boolean`. So technically, from a component API standpoint, you could have `isDefault` and `isPrimary`.

> Put another way, are we intentionally discouraging usage of raised + borderless?
That was intentional. During dev in Storybook, I noticed that the `<Card />` looked awkward with the border and the shadow. **But**, that is subjective. I can see the value of having finer grain control. 

So `variant="borderless | raised"` becomes:

`isBorderless` and `isRaised`.

> Should variant become something like border-style = border | raised | none?
Oo! I like `borderStyle` a lot more! It's more descriptive :). That would free up the `variant` prop to handle things like `shady`, as you mentioned! ❤️ 

If that sounds about right, I can make the change! Thanks again

Reply:
I tend to agree that cards that are both borderless _and_ raised look strange, so I'd be happy to keep them together under a single prop like `borderStyle` rather than making them independent. The way components are defined can encourage appropriate use:

`borderStyle = "borderless" | "raised" | undefined`

> That would free up the variant prop to handle things like shady, as you mentioned!

`variant` in general concerns me because it is so general and I think I'd continue to avoid it. Surely it makes sense in some cases, but it sounds like a prop that changes something _essential_ or _fundamental_ about a component. That doesn't seem like the case here.

Reply:
👍 The slightly tricky thing is that "raised" isn't really a border.. since it adds a shadow.

Would love to get some design thoughts on this! cc'ing @karmatosed / @melchoyce / @mapk  😊

Reply:
One situation where you might both elevation and border is when a card with border is drag'n'dropped to a new place. You'll want to apply elevation during the move over other elements.

Reply:
Another situation is when you might want to temporarily highlight/focus on something that normally has border. Example when linked to comment in Github:

![image](https://user-images.githubusercontent.com/87168/67070977-11ec7200-f18a-11e9-8515-34d1a096268f.png)

Tho arguably focus-border might be different mechanism but the elevation is one way to focus on things in grids of cards.

Reply:
@simison

> One situation where you might both elevation and border is when a card with border is drag'n'dropped to a new place. You'll want to temporarily apply elevation during the move over other elements.

> temporarily highlight/focus on something

Yes! I agree that both are valid cases :). The interesting thing about a Card component is just how many variations and potentially combined styles they can have. Similar to something like `<Button />`.

@simison + @sirreal

Apologies for my tardiness! I've been heads down on another project 😊 .

Just wanted to follow up on the border/variant APIs! What do you folks think we should go with for now? There are some wonderful suggestions above ☝️ 

I think we can make refinements later. I feel like it will be clearer when we refactor and use using the `<Card />` component, either in blocks or to replace UI in other components like `<Panel />`.

Thank you!!

Reply:
I can summarize my position as:
- I'd discourage "catch-all" props like `variant`
- I'd encourage independent props as much as makes sense. I believe that will keep things simpler and clearer over time (`borderStyle`, `elevation`, etc.).
- I don't have great naming suggestions 🙂 

I'd ask that you keep the above in mind, but no blockers.